spec/services/request_password_reset_spec.rb

```diff
@@ -3,15 +3,11 @@
 describe RequestPasswordReset do
   describe '#perform' do
     context 'when the user is not found' do
-      it 'sends the account does not exist email' do
+      it 'sends the account registration email' do
         email = 'nonexistent@example.com'
-
-        mailer = instance_double(ActionMailer::MessageDelivery, deliver_later: true)
-        allow(UserMailer).to receive(:account_does_not_exist).
-          with(email, 'request_id').and_return(mailer)
-        expect(mailer).to receive(:deliver_later)
-
-        RequestPasswordReset.new(email, 'request_id').perform
+        expect_any_instance_of(User).to receive(:send_custom_confirmation_instructions)
```

Comment:
Would it make sense to create a new email for this scenario? It might be confusing to get an email confirming account creation when the user was trying to reset their password. Maybe we should come up with new wording that says something like "you were trying to reset your password, but you don't have an account yet, so we created one for you."